nb_conda_kernels/manager.py

```diff
@@ -117,29 +119,61 @@ def _conda_info(self):
             relatively expensive.
         """
 
+        def get_conda_info_data():
+          # This is to make sure that subprocess can find 'conda' even if
+          # it is a Windows batch file---which is the case in non-root
+          # conda environments.
+          shell = CONDA_EXE == 'conda' and sys.platform.startswith('win')
+          try:
+            # conda info --json uses the standard JSON escaping
+            # mechanism for non-ASCII characters. So it is always
+            # valid to decode here as 'ascii', since the JSON loads()
+            # method will recover any original Unicode for us.
+            out = subprocess.check_output([CONDA_EXE, "info", "--json"],
+                                          shell=shell).decode('ascii')
+            conda_info = json.loads(out)
+            return conda_info, None
+          except Exception as err:
+            return None, err
+          finally:
+             self.wait_for_child_processes_cleanup()
```

Comment:
This method was added in #231 / #233 to resolve an issue where nb_conda_kernels would allow the accumulation of zombie processes under certain circumstances. It makes sense for the thread to take care of this.